src/WCSimAmBeGen.cc

```diff
@@ -0,0 +1,127 @@
+#include "WCSimAmBeGen.hh"
+#include "WCSimPrimaryGeneratorMessenger.hh"
+#include "WCSimPrimaryGeneratorAction.hh"
+
+#include "G4ParticleGun.hh"
+#include "G4PhysicalConstants.hh"
+#include "G4RandomDirection.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4SPSEneDistribution.hh"
+#include "G4SPSRandomGenerator.hh"
+#include "G4ParticleTable.hh"
+#include "G4Neutron.hh"
+#include "G4Gamma.hh"
+
+#include <random>
+
+using namespace std;
+
+WCSimAmBeGen::WCSimAmBeGen(){
+  // Initialise
+  this->Initialise();
+  wcsimdir = string(getenv("WCSIMDIR"))+"/data/";
+}
+
+WCSimAmBeGen::~WCSimAmBeGen(){
+  
+  // This needed to be deleted
+  delete myAmBeGun;
+  delete nEnergyDist;
+  delete rGen;
+}
+
+void WCSimAmBeGen::Initialise(){
+    nEnergyDist = new G4SPSEneDistribution();
+    rGen        = new G4SPSRandomGenerator();
+    myAmBeGun   = new G4ParticleGun();
+}
+
+G4double WCSimAmBeGen::GammaEnergy(){
+    
+    // Initialise Geant4 random number generator
+    G4Random::setTheEngine(new CLHEP::RanecuEngine);
+
+    // Define arrays
+    G4double probabilities[] = {0.26, 0.65, 0.08};
+    G4double energies[] = {0.0, 4.4 * MeV, 7.7 * MeV};
```

Comment:
you have another `energies` variable defined in .hh by not used. Remember to remove those ununsed variables

Reply:
👍